[#4081] Update to bootstrap 5

christinach opened the pull request:
closes #4081

christinach commented:
The next step for me is to update the dropdown buttons so that the advanced search  component works as expected.

christinach commented:
The color of the skip links is off but it shouldn't be a blocker.

christinach commented:
- @kevinreiss noticed that in the record page the show tools should be on the right side and not wrap. (Not blocker)

- cancel button in modal does not work.

sandbergja commented on Gemfile:
```diff
@@ -75,7 +76,7 @@ gem 'rsolr'
 gem 'rspec-rails'
 gem 'rubyzip', '>= 1.2.2'
 # Use SCSS for stylesheets
-gem 'sass-rails', '~> 6.0'
```
Could we remove this outright, rather than commenting it?

christinach replied:
yes I will remove it. I will add it in the commit for the Gemfile. I want to keep them tidy and clean.

christinach replied:
Thanks for looking into the skip links. I will check what you pointed out.

sandbergja commented on app/assets/stylesheets/blacklight.scss:
```diff
@@ -1,11 +1,4 @@
-$brand-success: #3B863B;
-$gray-light: #757575;
 
-@import 'bootstrap';
```
Thanks for organizing these better!

sandbergja commented on app/components/orangelight/search_bar_component.html.erb:
```diff
@@ -1,7 +1,7 @@
   <%= render Blacklight::HiddenSearchStateComponent.new(params: @params) %>
   <% if search_fields.length > 1 %>
-    <%= f.label :search_field, scoped_t('search_field.label'), class: 'sr-only visually-hidden' %>
+    <%= f.label :search_field, scoped_t('search_field.label'), class: 'visually-hidden visually-hidden' %>
```
Looks like the same class is here twice

```suggestion
    <%= f.label :search_field, scoped_t('search_field.label'), class: 'visually-hidden' %>
```

christinach replied:
oh yes! probably when I replaced sr-only with visually-hidden.

sandbergja commented on app/views/advanced/_numismatics_advanced_search_form.html.erb:
```diff
@@ -0,0 +1,62 @@
+<%= form_tag search_catalog_path, :class => 'advanced form-horizontal', :method => :get do  %>
```
How is this partial used? Does it replace the existing numismatics search form component?

christinach replied:
I'm not sure I will look into this one.

christinach replied:
@sandbergja the partial was removed while I was away. Rebasing the branch fixed it.

sandbergja reviewed (changes requested):
Thanks for all your work on this, @christinach 🐬 !  It was a big project, and it will be so great to be on Bootstrap 5!

* I think the skip links should be a blocker, I think that's a serious accessibility regression.  I think it might not be the color; in my devtools I see a 
```
position: absolute !important;
clip: rect(1px, 1px, 1px, 1px);
```
when I uncheck them I can see the skip links again.

* I see a very small difference in height in the search bar in the keyword dropdown vs. the search input:
![the search input is slightly taller than the keyword dropdown](https://github.com/user-attachments/assets/634e664a-d9ba-4aa9-9b90-433ea12b3541)

christinach commented:
@kevinreiss this branch is rebased against current main.

christinach commented:
> Thanks for all your work on this, @christinach 🐬 ! It was a big project, and it will be so great to be on Bootstrap 5!
> 
> * I think the skip links should be a blocker, I think that's a serious accessibility regression.  I think it might not be the color; in my devtools I see a
> 
> ```
> position: absolute !important;
> clip: rect(1px, 1px, 1px, 1px);
> ```
> 
> when I uncheck them I can see the skip links again.
> 
> * I see a very small difference in height in the search bar in the keyword dropdown vs. the search input:
>   ![the search input is slightly taller than the keyword dropdown](https://private-user-images.githubusercontent.com/7086943/385988214-634e664a-d9ba-4aa9-9b90-433ea12b3541.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3MDYxNjEsIm5iZiI6MTczMTcwNTg2MSwicGF0aCI6Ii83MDg2OTQzLzM4NTk4ODIxNC02MzRlNjY0YS1kOWJhLTRhYTktOWI5MC00MzNlYTEyYjM1NDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMjEyNDIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDBmNGFhODVjNzNmMDBjZDg2OTJiZTcxNjAwOGY4MjU0ODM3NGFkZmUzOTQ2ZTYyYjQ4NTg0MjcwODY5NWJmOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fJRTGcHzOFT2zkHgxPU1kVFE38NL5irQn9GQQ9b5Y68)

@sandbergja I rebased the bootstrap 5 branch on main which does not have the blacklight 8 upgrade. We reverted it yesterday. 
I don't see the skip links issue after rebasing.

I also don't see the '...small difference in height in the search bar in the keyword dropdown vs. the search input....'. See the screenshot below:
![search bar and dropdown](https://github.com/user-attachments/assets/b744fc59-faff-4ba5-82e4-06d5ca508d9e)

kevinreiss commented:
I noticed that Facet values in QA are underlined. 
<img width="413" alt="Screenshot 2024-11-18 at 8 15 40 AM" src="https://github.com/user-attachments/assets/b9b90523-dab1-465e-9ca4-ae9640721eab">

sandbergja commented:
@christinach the small difference between the search bar in the keyword dropdown vs. the search input looks like it's only in Firefox (and more subtly in Safari) -- I don't see it in Edge or Chrome.

Another subtle thing: If you go to "Your Account" > "Library Account", the horizontal rule is on top of the text "or".   Do we even need that horizontal rule, I wonder?

![the screen gives a choice between logging in with netID or alma account.  Between them is the word or with a line on top of it](https://github.com/user-attachments/assets/e2e7aac1-616b-46b1-b4da-ee7fffca531b)

christinach commented:
> @christinach the small difference between the search bar in the keyword dropdown vs. the search input looks like it's only in Firefox (and more subtly in Safari) -- I don't see it in Edge or Chrome.
> 
> Another subtle thing: If you go to "Your Account" > "Library Account", the horizontal rule is on top of the text "or". Do we even need that horizontal rule, I wonder?
> 
> ![the screen gives a choice between logging in with netID or alma account. Between them is the word or with a line on top of it](https://private-user-images.githubusercontent.com/7086943/389738720-e2e7aac1-616b-46b1-b4da-ee7fffca531b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI2MzA0MDYsIm5iZiI6MTczMjYzMDEwNiwicGF0aCI6Ii83MDg2OTQzLzM4OTczODcyMC1lMmU3YWFjMS02MTZiLTQ2YjEtYjRkYS1lZTdmZmZjYTUzMWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjZUMTQwODI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTlhOWU5YjQzN2QxNzIzMWFhNGM1Mzk4YzUwYTBjYTcxMzY5MzNiZGU5MDM0MWZmZDEzMDFmYjY1YzNjMTdiMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gNPdWfNl4ezaXvptpF7v-cBh4GDamKH-cUtbb_pXmac)

@sandbergja thanks. I fixed the height discrepancy. I missed an input-group-prepend which was causing that.
I don't think we need the separator line. I will remove it in another PR against main.

sandbergja reviewed (approved):
Thanks, @christinach 🐬 !  This is so great!
